site/en/blog/navigateevent-intercept/index.md

```diff
@@ -0,0 +1,112 @@
+
+---
+title: Changes to NavigateEvent in Chrome 105
+description: >
+  The Navigation API gets two new methods on NavigateEvent: intercept() replaces transitionWhile() and scroll() replaces restoreScroll().
+layout: 'layouts/blog-post.njk'
+date: 2022-08-04
+tags:
+  - chrome-105
+  - capabilities
+hero: 'image/sQ51XsLqKMgSQMCZjIN0B7hlBO02/CdDfdtqJPPTeVCd3QqRR.jpeg'
+alt: >
+  A saiboat navigating the big blue waters.
+authors:
+  - joemedley
+---
+
+Chrome 105 introduces two new methods on the `NavigateEvent` of the Navigation API (introduced in 102) to improve on methods that have proved problematic in practice. `Intercept()`, which let's developers control the state following the navigation, replaces `transitionWhile()`, which proved difficult to use. The `scroll()` method, which scrolls to an anchor specified in the URL, replaces `restoreScroll()` which does not work for all types of navigation. 
+
+In this article, I'll explain the problems with both and how the new methods fix those problems.
+
+## NavigateEvent.transitionWhile()
+
+The `NavigateEvent.trasitionWhile()` method, introduced with the Navigation API in Chrome 102, intercepts navigation for client-side transitions in single page apps. Its first argument is a promise that signals to the browser and to other parts of the web application that it's finished.
+
+This worked poorly in practice. Consider this common coding pattern:
+
+```js
+event.transitionWhile((async () => {
+  doSyncStuff();
+  await doAsyncStuff();
+})());
+```
+
+This is functionally equivalent to the code below. It causes some portion of the navigation to run before the navigation is even finished.
+
+```js
+doSyncStuff();
+event.transitionWhile((async () => {
+  await doAsyncStuff();
+})());
+```
+
+One example where this can mess up an app is in scroll restoration logic where it acts on the DOM before the navigation transition instead of after. 
+
+### What's changed
+
+To replace `transitionWhile()`, the current spec introduces `NavigateEvent.intercept()`. The new method takes a handler in addition to the `focusReset` and `scrollRestoration` properties supported by `transitionWhile()`. The new handler always runs after the navigation completes, avoiding the problems with `transitionWhile()`.
+
+The `transitionWhile()` method is still available, but it has been deprecated and will be removed in Chrome 108.
+
+### Using intercept()
+
+The `NavigateEvent.intercept()` cannot be used on all events. If the current `Document` object has a URL that can be rewritten to a destination URL, `intercept()` may be called. You can't call it on cross-document back/forward navigations. Doing so will throw a [`DOMException`](https://developer.mozilla.org/docs/Web/API/DOMException) of the `"SecurityError"` type.
```

Comment:
```suggestion
The `NavigateEvent.intercept()` has the same restrictions as `transitionWhile()`, as in it cannot be called on all navigation events. Cross-origin navigations cannot be intercepted, neither can cross-document traversals. Doing so will throw a [`DOMException`](https://developer.mozilla.org/docs/Web/API/DOMException) of the `"SecurityError"` type.
```

When I first read this, I thought it was suggesting that `intercept()` has more restrictions than `transitionWhile()`.